vulture/core.py

```diff
@@ -53,6 +53,16 @@ def _is_test_file(filename):
     )
 
 
+def _assigns_special_variable__all__(node):
+    assert isinstance(node, ast.Assign)
+    name_targets = {
+        target.id for target in node.targets if isinstance(target, ast.Name)
+    }
+    return "__all__" in name_targets and isinstance(
+        node.value, (ast.List, ast.Tuple)
+    )
```

Comment:
I can imagine that the following versions runs slightly faster, since it avoids a temporary set and makes the cheap if-test first:

```suggestion
def _assigns_special_variable__all__(node):
    assert isinstance(node, ast.Assign)
    return (
        isinstance(node.value, (ast.List, ast.Tuple)) and
        any(target.id == "__all__" for target in node.targets if isinstance(target, ast.Name))
    )
```

Reply:
I agree that it runs faster, but in my opinion it is slightly less readable.

But of course I am fine with it ;)